### PR TITLE
pd: fix a dead lock

### PR DESCRIPTION
**Pull request by overvenus**
Read lock won't be dropped inside of the `match` block.

### CODE REVIEW

File: src/pd/util.rs [hunk 1]
```diff
@@ -285,7 +285,9 @@ where
     F: Fn(&PdClient) -> GrpcResult<R>,
 {
     for _ in 0..retry {
-        match func(&client.inner.rl().client).map_err(Error::Grpc) {
+        // DO NOT put any lock operation in match statement, or it will cause dead lock!
+        let ret = { func(&client.inner.rl().client).map_err(Error::Grpc) };
```
Comment (BusyJay) [inline src/pd/util.rs]: Any case to cover this?

**Comment by siddontang** [comment]
I remember that we meet the similar deadlock many times, we should fully check our code to see whether we use the same way for the lock. @overvenus

**Comment by overvenus** [comment]
There are still some "match-lock", but they are all in tests, I think they won't cause any trouble.

Try `grep -Ersn 'match .*(wl|rl|read|write|lock)\(\).*'` for more details.

File: tests/pd/mock/mocker/retry.rs [hunk 2]
```diff
@@ -32,16 +35,36 @@ impl Retry {
             count: AtomicUsize::new(0),
         }
     }
+
+    fn ok_or_error(&self) -> bool {
```
Comment (BusyJay) [inline tests/pd/mock/mocker/retry.rs]: Confusing method name.

**Comment by siddontang** [comment]
@overvenus 

This potential deadlock in the test may cause test failed, we should fix them later. 

LGTM

**Pull request merged**